### ISSUE TITLE
How to change the value by code?

### ISSUE THREAD

**Issue by hlblade** [stla/shinyDatetimePickers#17]
Thx

**Comment by stla** [comment]
Hello,
Currently the `update*` functions are not implemented. I don't remember why I didn't implement them. Maybe because I was  not able to do so... I will take a look.

**Comment by hlblade** [comment]
Hi, Your package is the best datetimepicker I can found for shiny. Hope you can add this feature.Thx in advanced!Leon-------- 原始邮件 --------发件人： stla ***@***.***>日期： 2024年2月4日周日 半夜11:34收件人： stla/shinyDatetimePickers ***@***.***>抄送： hlblade ***@***.***>, Author ***@***.***>主    题： Re: [stla/shinyDatetimePickers] How to change the value by code? (Issue #17)
Hello,
Currently the update* functions are not implemented. I don't remember why I didn't implement them. Maybe because I was  not able to do so... I will take a look.

**Comment by stla** [comment]
I've managed too add this feature for the `datetimePickerInput`. I din't manage yet for the `datetimeSliderPickerInput`, and I didn't try yet for the `datetimeMaterialPickerInput`. Here is an example for the `datetimePickerInput`. If you want to try it, you have to install the `updatefunctions` branch:
```r
remotes::install_github("stla/shinyDatetimePickers@updatefunctions")
```
The example:
```r
library(shinyDatetimePickers)
library(shiny)

ui <- fluidPage(
  br(),
  sidebarLayout(
    sidebarPanel(
      tags$fieldset(
        tags$legend("Click to change time"),
        datetimePickerInput(
          "dtpicker",
          style =
            "font-family: Montserrat, 'Segoe UI', Tahoma, sans-serif;"
        )
      ),
      actionButton("update", "Update")
    ),
    mainPanel(
      verbatimTextOutput("dtpicker")
    )
  )
)

server <- function(input, output, session){
  output[["dtpicker"]] <- renderPrint({
    input[["dtpicker"]]
  })

  observeEvent(input[["update"]], {
    updateDatetimePickerInput(session, "dtpicker", Sys.time())
  })
}

shinyApp(ui, server)
```

**Comment by hlblade** [comment]
Thx very much for your swift reply and great solution. I will have a try.-------- 原始邮件 --------发件人： stla ***@***.***>日期： 2024年2月5日周一 凌晨4:32收件人： stla/shinyDatetimePickers ***@***.***>抄送： hlblade ***@***.***>, Author ***@***.***>主    题： Re: [stla/shinyDatetimePickers] How to change the value by code? (Issue #17)
I've managed too add this feature for the datetimePickerInput. I din't manage yet for the datetimeSliderPickerInput, and I didn't try yet for the datetimeMaterialPickerInput. Here is an example for the datetimePickerInput. If you want to try it, you have to install the updatefunctions branch:
***@***.***")
The example:
library(shinyDatetimePickers)
library(shiny)

ui <- fluidPage(
  br(),
  sidebarLayout(
    sidebarPanel(
      tags$fieldset(
        tags$legend("Click to change time"),
        datetimePickerInput(
          "dtpicker",
          style =
            "font-family: Montserrat, 'Segoe UI', Tahoma, sans-serif;"
        )
      ),
      actionButton("update", "Update")
    ),
    mainPanel(
      verbatimTextOutput("dtpicker")
    )
  )
)

server <- function(input, output, session){
  output[["dtpicker"]] <- renderPrint({
    input[["dtpicker"]]
  })

  observeEvent(input[["update"]], {
    updateDatetimePickerInput(session, "dtpicker", Sys.time())
  })
}

shinyApp(ui, server)

**Comment by stla** [comment]
Now the same feature is available for the `datetimeSliderPickerInput`.  I'll implement it for the `datetimeMaterialPickerInput` tomorrow, which should not cause any difficulty.

**Comment by hlblade** [comment]
Hi,
The update function works very well on my side too. Thx very much for your great work.
By the way, how to change the font size? 

The following code didn't work.

datetimeMaterialPickerInput(

          "dtpicker",

          style =

            "font-size:12px;"

        )




Best wishes,
Leon




At 2024-02-05 12:19:21, "stla" ***@***.***> wrote:

Now the same feature is available for the datetimeSliderPickerInput. I'll implement it for the datetimeMaterialPickerInput tomorrow, which should not cause any difficulty.

**Comment by stla** [comment]
Indeed, there's already a font-size property set on the picker and it has priority. You can try to include this CSS in the app:
```css
.MuiInputBase-input {
  font-size: 12px;
}
```
I didn't try.

**Comment by hlblade** [comment]
It works well, thanks very much.-------- 原始邮件 --------发件人： stla ***@***.***>日期： 2024年2月6日周二 晚上8:19收件人： stla/shinyDatetimePickers ***@***.***>抄送： hlblade ***@***.***>, Author ***@***.***>主    题： Re: [stla/shinyDatetimePickers] How to change the value by code? (Issue #17)
Indeed, there's already a font-size property set on the picker and it has priority. You can try to include this CSS in the app:
.MuiInputBase-input {
  font-size: 12px;
}
I didn't try.
